vm/option.go

```diff
@@ -11,45 +11,57 @@ import (
 	"github.com/ava-labs/hypersdk/gossiper"
 )
 
-type Option func(*VM) error
+type OptionFunc func(*VM, []byte) error
 
-func WithManualBuilder() Option {
-	return func(vm *VM) error {
+type Option struct {
+	Namespace  string
+	OptionFunc OptionFunc
+}
+
+func NewOption(namespace string, optionFunc OptionFunc) Option {
+	return Option{
+		Namespace:  namespace,
+		OptionFunc: optionFunc,
+	}
+}
+
+func WithManualBuilder() OptionFunc {
```

Comment:
There should be two different concepts - an option registers a useful service ie. `indexer.With` registers the indexer service. The functions here are meant to set a field on the VM and can be called from `OptionFunc`s defined by services, but don't need the `configBytes` here since they will never be treated as an `OptionFunc` directly.

In the case of `WithManualBuilder` and `WithManualGossiper` these should really be `WithBuilder` and `WithGossiper` and `WithManual` would call both to register the manual gossiper and builder.